src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -2713,7 +2713,7 @@ bool primitive_inst::is_valid_fusion() const {
                                                          cldnn::format::dimension(data_layout.format),
                                                          false);
 
-            if (gemm_dims[0] != data_dims[0])
+            if (gemm_dims[0] != data_dims[0] && gemm_dims[1] != 1)
```

Comment:
Shouldn't this be more generic like this?
```suggestion
            if (gemm_dims[0] * gemm_dims[1] != data_dims[0] * data_dims[1])
```